src/components/search-by-demographics/search-by-demographics.component.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from "react";
+
+import {
+  DatePicker,
+  DatePickerInput,
+  Column,
+  NumberInput,
+  Switch,
+  ContentSwitcher,
+} from "carbon-components-react";
+import dayjs from "dayjs";
+import { useTranslation } from "react-i18next";
+
+import { SearchByProps } from "../../types";
+import SearchButtonSet from "../search-button-set/search-button-set";
+import styles from "./search-by-demographics.style.scss";
+import {
+  getDescription,
+  getQueryDetails,
+} from "./search-by-demographics.utils";
+
+const SearchByDemographics: React.FC<SearchByProps> = ({
+  onSubmit,
+  isLoading,
```

Comment:
You don't need a prop to check the loading state.

 Just wait until `onSubmit` is resolved.